### PR TITLE
MSA-2084: Spruce Sensor DTH new model and health check

### PR DESCRIPTION
**Pull request by natec007**
Submitting 'Spruce Sensor DTH new model and health check' for publication

### CODE REVIEW

File: devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy [hunk 1]
```diff
@@ -14,25 +14,34 @@
...
 		
 		capability "Configuration"
 		capability "Battery"
         capability "Relative Humidity Measurement"
         capability "Temperature Measurement"
         capability "Sensor"
+        capability "Health Check"
         //capability "Polling"
 		
+        attribute "checkInterval", "string"
```
Comment (jackchi) [inline devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy]: No need to add `checkInterval` here. It's actually included as a `NUMBER` in the Capbility `Health Check`

File: devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy [hunk 2]
```diff
@@ -293,6 +302,11 @@ def setConfig(){
     sendEvent(name: 'configuration',value: configInterval, descriptionText: "Configuration initialized")
 }
 
+def installed(){
+	//check every 1 hour + 2mins
+    sendEvent(name: "checkInterval", value: 1 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```
Comment (jackchi) [inline devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:306]: Let's put this `sendEvent("checkInterval")` in both `installed()` & `updated()`

**Comment by jackchi** [comment]
Is this DTH "pingable()" - meaning, that we can contact the device and device would generate an event in the parse() method?

**Review by jackchi — CHANGES_REQUESTED** [review]
1. Take out `checkInterval` from custom attribute
2. Add `sendEvent("checkInterval")` to both updated/installed
3. Can this device be pinged?

File: devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy (hunk 2 above)
Comment (natec007) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:306]: Its a sleepy ZigBee device, so can't be pinged except when awake.
Made the changes requested

Comment (jackchi) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:306]: So how often does this device check-in (generate an event) when idle? Currently if our platform doesn't hear from this device in 62min (your checkInterval value) it'll mark it `OFFLINE`

Comment (natec007) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:306]: It pings the coordinator (hub) every 10 mins but reports data between 10-120mins.  So if the ZigBee event is caught then 62mins should be fine, if you only look at reported data, it should change to 2 hours.

Comment (jackchi) [reply devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:306]: 👍

**Review by jackchi — APPROVED** [review]
Everything looks good.

**Pull request merged**